lib/Doctrine/DBAL/Connection.php

```diff
@@ -1589,11 +1589,17 @@ public function ping()
             return $this->_conn->ping();
         }
 
+        set_error_handler(function ($severity, $message) {
+            throw new \PDOException($message, $severity);
```

Comment:
Given that this is a PHP bug, I think it would be better to work around this in PDOMySQL driver, and revert when the PHP bug is fixed (probably a long time)